src/providers/UserStateProvider.tsx

```diff
@@ -158,7 +158,7 @@ function WrappedUserStateProvider({
     <userStateContext.Provider value={{ ...state, ...interactors, dispatch }}>
       {children}
       {(state.config.userEmail === null || state.config.groupId === null) && (
```

Comment:
Should `whereFound` be on this list?

Maybe we should put this into a memo or something -- the condition is sort of complex to just shove inline like this (I know this is my code 👼)

Reply:
I added the condition, could you help me with the memoization step? I'm confused on where to declare the memo of it all.

Reply:
Maybe this should just be a constant defined earlier in the function.

Eg. add this earlier in the func near where you are doing hooks n stuff
```
const userNeedsSetup = (state.config.userEmail === null || state.config.groupId === null)
```
Then replace the inline condition with
```
(userNeedsSetup && (...Modal...))
```

Reply:
I've made this change to the above suggestion. It might also be useful to define an array which defines necessary information from the user (ie, state.config.userEmail, state.config.groupId, etc.) and just looping threw those elements.